aesd-char-driver/main.c

```diff
@@ -150,15 +154,18 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
   }
 
   // No return check for mutex unlock //
+  device->buf_size += bytesWrittenToKernelSpace;
```

Comment:
@Vignesh-Vadivel it looks like you are missing the logic to adjust the buffer size down as elements are removed from the circular buffer?